src/server/worker/master.go

```diff
@@ -713,6 +713,9 @@ func (a *APIServer) waitJob(pachClient *client.APIClient, jobInfo *pps.JobInfo,
 			}
 		}
 		if jobInfo.EnableStats {
+			if jobInfo.StatsCommit == nil {
+				return fmt.Errorf("stats are enabled, but the job has a nil stats commit")
+			}
```

Comment:
Do we actually want to error out here? It seems like based on the code below it wouldn't be too hard to just skip the stats commit and still finish the job. In general we try to avoid stats errors failing jobs and keep stats as non-critical.

Reply:
makes sense